test/e2e/lazyload_test.go

```diff
@@ -0,0 +1,320 @@
+	serviceFenceYaml := readFile(test, "samples/lazyload/servicefence_productpage.yaml")
+	framework.RunKubectlOrDieInput(nsApps, serviceFenceYaml, "create", "-f", "-")
+	defer func() {
+		testResourceToDelete = append(testResourceToDelete, &TestResource{Namespace: nsApps, Contents: serviceFenceYaml})
+	}()
+
+	// check
+	svfGvr := schema.GroupVersionResource{
+		Group:    svfGroup,
+		Version:  svfVersion,
+		Resource: svfResource,
+	}
+
+	svfCreated := false
+	for i := 0; i < 60; i++ {
+		_, err := f.DynamicClient.Resource(svfGvr).Namespace(nsApps).Get(context.TODO(), svfName, metav1.GetOptions{})
+		if err != nil {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		svfCreated = true
+		break
+	}
+	if svfCreated != true {
+		framework.Failf("Failed to create servicefence.\n")
+	}
+
+	sidecarGvr := schema.GroupVersionResource{
+		Group:    sidecarGroup,
+		Version:  sidecarVersion,
+		Resource: sidecarResource,
+	}
+
+	sidecarCreated := false
+	for i := 0; i < 60; i++ {
+		_, err := f.DynamicClient.Resource(sidecarGvr).Namespace(nsApps).Get(context.TODO(), sidecarName, metav1.GetOptions{})
+		if err != nil {
+			time.Sleep(500 * time.Millisecond)
+			continue
+		}
+		sidecarCreated = true
+		break
+	}
+	if sidecarCreated != true {
+		framework.Failf("Failed to create sidecar.\n")
+	}
+
+	ginkgo.By("serviceFence and Sidecar create successfully")
+}
+
+func updateSidecar(f *framework.Framework) {
+	sidecarGroup := "networking.istio.io"
+	sidecarVersion := "v1beta1"
+	sidecarResource := "sidecars"
+	sidecarName := "productpage"
+
+	pods, err := f.ClientSet.CoreV1().Pods(nsApps).List(context.TODO(), metav1.ListOptions{})
+	framework.ExpectNoError(err)
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "ratings") {
+			for i := 0; i < 10; i++ {
+				_, _, err = f.ExecShellInPodWithFullOutput(pod.Name, nsApps, "curl \"productpage:9080/productpage\"")
+				framework.ExpectNoError(err)
+			}
+		}
+	}
+
+	sidecarGvr := schema.GroupVersionResource{
+		Group:    sidecarGroup,
+		Version:  sidecarVersion,
+		Resource: sidecarResource,
+	}
+	sidecarUpdated := false
+Loop:
+	for i := 0; i < 120; i++ {
+		sidecar, err := f.DynamicClient.Resource(sidecarGvr).Namespace(nsApps).Get(context.TODO(), sidecarName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		egress, _, err := unstructured.NestedSlice(sidecar.Object, "spec", "egress")
+		framework.ExpectNoError(err)
+		hosts, _, err := unstructured.NestedStringSlice(egress[0].(map[string]interface{}), "hosts")
+		framework.ExpectNoError(err)
+		for _, host := range hosts {
+			// TODO Remove global-sidecar condition after 503 UH problem is solved
+			if strings.Contains(host, "details") || strings.Contains(host, "reviews") || strings.Contains(host, "global-sidecar") {
+				sidecarUpdated = true
+				break Loop
+			}
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	if !sidecarUpdated {
+		framework.Failf("sidecar updated failed\n")
+	}
+	ginkgo.By("sidecar updated successfully")
```

Comment:
we also need to confirm that the traffic will not be routed to global sidecar while sidecar cr is created and effective.

Reply:
Ok. I will consider using istio-proxy access log or related prometheus data change to verify this result.